lib/internal/Magento/Framework/Mail/Template/TransportBuilder.php

```diff
@@ -279,4 +279,20 @@ protected function prepareMessage()
 
         return $this;
     }
+
+    /**
+     * Set mail from address by store.
+     *
+     * @param string|array $from
+     * @param string|int $store
+     *
+     * @return $this
+     */
+    public function setFromByStore($from, $store)
```

Comment:
I would suggest to avoid adding new public methods when not necessary. Every public method is a kind of contract for backward compatibility.

Changing `setFrom` and adding an optional parameter could be a better option.

See https://github.com/magento/magento2/pull/16356/files trying to fix the same issue.

Reply:
I am facing the same issue , when a customer placed an order then customer is getting the mail but not admin.. And the second thing is that it is the taking the different host as my id is socketswitches@socket-switches.com but system is sending from socketswitches@ssdvps-80478.vps.net  . ssdvps-80478.vps.net this is the server ..Any help!!!

Reply:
I had tried this code but isn't working.

Reply:
I am also facing same issue.. can you help in this ?